pkg/xray/resource_xray_repository_config.go

```diff
@@ -0,0 +1,349 @@
+				"index_new_artifacts": pattern.IndexNewArtifacts,
+				"retention_in_days":   pattern.RetentionInDays,
+			}
+
+			ps = append(ps, p)
+		}
+
+		return ps
+	}
+
+	var packRepoPathsConfigList = func(repoPathsConfig *RepoPathsConfiguration) []interface{} {
+		if repoPathsConfig == nil {
+			return []interface{}{}
+		}
+
+		m := map[string]interface{}{}
+		m["pattern"] = packPatterns(repoPathsConfig.PatternList)
+		m["all_other_artifacts"] = packAllOtherArtifacts(repoPathsConfig.OtherArtifacts)
+
+		return []interface{}{m}
+	}
+
+	var packRepositoryConfig = func(ctx context.Context, repositoryConfig RepositoryConfiguration, d *schema.ResourceData) diag.Diagnostics {
+		if err := d.Set("repo_name", repositoryConfig.RepoName); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("repo_config", packGeneralRepoConfig(repositoryConfig.RepoConfig)); err != nil {
+			return diag.FromErr(err)
+		}
+		if err := d.Set("repo_paths_config", packRepoPathsConfigList(repositoryConfig.RepoPathsConfig)); err != nil {
+			return diag.FromErr(err)
+		}
+
+		return nil
+	}
+
+	var resourceXrayRepositoryConfigRead = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		repositoryConfig := RepositoryConfiguration{}
+
+		resp, err := m.(*resty.Client).R().SetResult(&repositoryConfig).
+			SetPathParams(map[string]string{
+				"repo_name": d.Id(),
+			}).
+			Get("xray/api/v1/repos_config/{repo_name}")
+
+		if err != nil {
+			if resp != nil && resp.StatusCode() != http.StatusOK {
+				tflog.Error(ctx, fmt.Sprintf("Request payload is invalid as repo (%s) is either not indexed or does not exist", d.Id()))
+				d.SetId("")
+			}
+			return diag.FromErr(err)
+		}
+
+		return packRepositoryConfig(ctx, repositoryConfig, d)
+	}
+
+	var resourceXrayRepositoryConfigCreate = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		repositoryConfig := unpackRepositoryConfig(d)
+
+		resp, err := m.(*resty.Client).R().SetBody(&repositoryConfig).Put("xray/api/v1/repos_config")
+		if err != nil {
+			if resp != nil && resp.StatusCode() != http.StatusOK {
+				tflog.Error(ctx, fmt.Sprintf("Request payload is invalid as repo (%s) is either not indexed or does not exist", d.Id()))
+			}
+			return diag.FromErr(err)
+		}
+
+		d.SetId(repositoryConfig.RepoName)
+		return resourceXrayRepositoryConfigRead(ctx, d, m)
+	}
+
+	// No delete functionality provided by API.
+	// Delete function will return a warning and remove the Id from the state.
+	// The option with restoring a default configuration is not viable, because the default can be changed.
+	var resourceXrayRepositoryConfigDelete = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		tflog.Warn(ctx, fmt.Sprintf("There is no delete dunctionality in the API, so the configuration is not "+
+			"removed from the Artifactory, but (%s) is removed from the Terraform state", d.Id()))
+		d.SetId("")
```

Comment:
Logging the warning is good but we also need to return a warning `Diagnostic` like: https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/resource/configuration/resource_artifactory_general_security.go#L59 so users will see the message.

Reply:
Added.